### ISSUE TITLE
Obsoleted terms have no label in JSON edition

### ISSUE THREAD

**Issue by pnrobinson** [monarch-initiative/mondo#889]
In HP, we have as label: "obsolete: previous label" 
This is useful for understanding what the term was and why is was obsoleted. 
This information is missing in MONDO  (many terms). Here is an example
```
{
      "id" : "http://purl.obolibrary.org/obo/MONDO_0004915",
      "meta" : {
        "basicPropertyValues" : [ {
          "pred" : "http://purl.obolibrary.org/obo/IAO_0000231",
          "val" : "http://purl.obolibrary.org/obo/IAO_0000227"
        }, {
          "pred" : "http://purl.obolibrary.org/obo/IAO_0100001",
          "val" : "http://purl.obolibrary.org/obo/MONDO_0005076"
        } ],
        "deprecated" : true
      },
      "type" : "CLASS"
    },
```

**Comment by cmungall** [comment]
This corresponds to a merge (alt_id in obo format). The structure above is isomorphic to the OWL. I don't think we'd want to inject a label where there is none, but we could have a more helpful tag at the json level to indicate merges

**Comment by pnrobinson** [comment]
OK, but it would be much more useful to retain a label like we do in the HPO. It is next to impossible to figure out where classes like this come from without knowing the unwritten rules

**Comment by matentzn** [comment]
So we have two Terms:

A "label 1"
B "label 2"

We merge them both into B "label 2"

@cmungall I also dont see why we couldnt just retain "obsolete label 1" on A? Thats not a format issue, this is an ontology maintainance decision, and I agree a bit with @pnrobinson here.

This does not however make the @cmungall suggestion redundant to add a better element for merges in obographs; I made a ticket here: https://github.com/geneontology/obographs/issues/68

**Comment by maglott** [comment]
if there were a decision to merge, i.e. A really is the same as B, why wouldn't "label 1" be an alternate label for B. 
If "label 1" is not valid for B now, why was A merged into B rather than A being made obsolete?

**Comment by matentzn** [comment]
Oh no, @maglott sorry, what you are saying will happen for sure. The question is only now whether in addition to making label 1 an alternate label for B, to get something like this (dont worry about syntax, just principle):

```
B rdfs:label "label 2"
B altLabel "label 1"

A rdfs:label "obsolete label 1"
A replacementId B
```

The point being that when someone looks at A, they can see more quickly what that used to be without having to click on the B to retrieve information on what it was replaced with..

**Comment by maglott** [comment]
Thanks for responding so quickly.  I am still quite a novice with syntax for managing ontologies.

**Comment by matentzn** [comment]
:) No worries at all! Thanks for engaging! :)

**Comment by nicolevasilevsky** [comment]
@matentzn do you know if this can be closed? If not, do you know what the action items are?
thanks!

**Comment by matentzn** [comment]
Obsolete classes have labels:

```
 {
      "id" : "http://purl.obolibrary.org/obo/MONDO_0000311",
      "meta" : {
        "basicPropertyValues" : [ {
          "pred" : "http://purl.obolibrary.org/obo/IAO_0100001",
          "val" : "MONDO:0018612"
        } ],
        "deprecated" : true
      },
      "type" : "CLASS",
      "lbl" : "obsolete congenital hypothyroidism"
    }, 
```

However, the term Peter selected does not have a label, because it has a replacement id, as per convention. There are [117 such terms in mondo](https://api.triplydb.com/s/j23P1GzZT).

_The question in this ticket is:_
Given that there is `colorectal adenocarcinoma` in Mondo (MONDO:0000528) and there is another term MONDO:0005008 that was deprecated and replaced by that term - would it be (A) confusing or (B) helpful to the user to see `obsolete colorectal adenocarcinoma` on MONDO:0005008?

@pnrobinson @cmungall

**Comment by pnrobinson** [comment]
In HPO we do just this. But we do not display obsolete terms in the browser.

**Comment by matentzn** [comment]
Yeah! I kinda agree. However, the question here is not so much about the browser, but the user of the ontology. Lets say they open it, in whatever browser tool they want, would it be more helpful to see the `obsolete colorectal adenocarcinoma` or would it confuse them?

**Comment by sabrinatoro** [comment]
This issue was created more than 2 years ago and has not been addressed to date. It has therefore been identified as a candidate to be closed without action.
If anyone believes that this issue should be addressed, please comment here.
Otherwise, this issue will automatically be closed with the October 2025 release.

Note: I think the Mondo users are used to the way we display obsolete terms with the label including "obsoleted - "

**Comment by matentzn** [comment]
This issue is low priority, but I can fix it in less than 5 minutes which makes it a candidate for fixing.

There are now 74 such cases: https://api.triplydb.com/s/wWaYApZi4

The only thing it needs is a decision:

Given a class MOND:123 that is deprecated and has _no label_, which is replaced by MONDO:124 which has a label _A_, does it make sense to add "obsolete A" as a label to MONDO:123?

**Comment by cmungall** [comment]
I don't get why we would do this. The current behavior is ultra-standard
for go, uberon, cl, hp, mp, ... and number of other ontologies. This has
been discussed a lot of multiple other tickets across multiple repos.
Making this change would change the obo format serialization (no more
alt_ids), it would introduce a confusing precedent not seen in any other
ontology, could potentially confuse users with strange non-name names,
breaking obo naming conventions, etc etc. If we want to officially undo
past Merge actions and turn then into ObsoletionWithReplacements then it
will be a bit more work as we have to go back to old versions...

We are in a bit of a weird in-between state right now though....

➜  git:(main) ✗ curl -L -s $OBO/mondo.obo | grep 'name: obsolete MONDO' | wc
      44     132    1276
➜  git:(main) ✗ curl -L -s $OBO/mondo.obo | grep alt_id | wc
      74     148    1628

**Comment by matentzn** [comment]
I now see your second 2019 comment - while I don't get the concern exactly (there will be many obsolete classes with labels, and only some without, and I am not arguing to change the representation pattern (which is _not_ altid)), its also not at all important right now - lets leave things as they are then (ticket can be closed)!
